src/api/api_missions.nit

```diff
@@ -35,6 +36,32 @@ end
 
 class APIMission
 	super MissionHandler
+	super AuthHandler
+
+	redef fun post(req, res) do
+		var player = get_player(req, res)
+		if player == null then return
+		var mission = get_mission(req, res)
+		if mission == null then return
+
+		var post = req.body
+
+		var deserializer = new JsonDeserializer(post)
+		var submission = new Submission.from_deserializer(deserializer)
+		if not deserializer.errors.is_empty then
+			res.error 400
+			print "Error deserializing submitted mission: {post}"
+			return
+		end
+
+		var runner = config.engine_map[submission.engine]
+		var program = new Program(player, mission, submission.source)
+		runner.run(program)
+
+		program.update_status(config)
```

Comment:
I think the stars should check the state of the program before updating instead of the other way around